### PR TITLE
Arm32

### PR DESCRIPTION
**Pull request by PaoloS02**
WIP to add support for ARM and the stm32f4-discovery board.

### CODE REVIEW

File: config/arm32/boards/stm32f4-discovery/board.cfg [hunk 1]
```diff
@@ -59,3 +59,17 @@
...
+    '-mfloat-abi=hard', '-mfpu=fpv4-sp-d16', '-mthumb'
+]
+
+ldflags = [
+    '-march=armv7e-m', '-mcpu=cortex-m4',
+    '-mfloat-abi=hard', '-mfpu=fpv4-sp-d16', '-mthumb',
+
+    '-Wl,-lc', '-Wl,-lnosys',
+    '-Wl,-Map=output.map',
+    '-T/home/paolo/projects/EMBENCH/embench-iot/config/arm32/boards/stm32f4-discovery/stm32f4-discovery.ld',
```
Comment (luismarques) [inline config/arm32/boards/stm32f4-discovery/board.cfg]: The paths shouldn't be hardcoded.

Comment (PaoloS02) [reply config/arm32/boards/stm32f4-discovery/board.cfg]: Thank you for spotting that. I needed to pass the work over to Jeremy quickly and I missed that. I'll fix it asap.

File: config/arm32/chips/size-test-gcc/chip.cfg [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# - board specific value
+# - command line value
+
+# For flags, this priority is applied to individual flags, not the complete
+# list of flags.
+
+cc = 'arm-none-eabi-gcc'
+cflags = [
+    '-c', '-Os',
+    '-fdata-sections', '-ffunction-sections',
```
Comment (claziss) [inline config/arm32/chips/size-test-gcc/chip.cfg]: GCC ARM is using section anchors (i.e., `-fsection-anchors`)  when compiling for size. Hence, the use of `-fdata-section` may lead to larger/slower output. A test for the best option choice needs to be performed for the best results.

Comment (jeremybennett) [reply config/arm32/chips/size-test-gcc/chip.cfg]: @claziss Many thanks. We'll take measurements to determine the best choice.

Comment (PaoloS02) [reply config/arm32/chips/size-test-gcc/chip.cfg]: Yes. The use -fdata-section actually generated a larger output. I removed it. Thank you for the tip!

**Comment by jeremybennett** [comment]
This has been subsumed by pull request #42, so this pull request is being closed. Thanks for your contribution.

**Pull request closed without merging**